runtime/lua/vim/treesitter/_meta.lua

```diff
@@ -62,19 +63,31 @@ function TSNode:_rawquery(query, captures, start, end_, opts) end
 ---@field _set_logger fun(self: TSParser, lex: boolean, parse: boolean, cb: TSLoggerCallback)
 ---@field _logger fun(self: TSParser): TSLoggerCallback
 
----@class TSTree
+---@class TSTree: userdata
 ---@field root fun(self: TSTree): TSNode
 ---@field edit fun(self: TSTree, _: integer, _: integer, _: integer, _: integer, _: integer, _: integer, _: integer, _: integer, _:integer)
 ---@field copy fun(self: TSTree): TSTree
 ---@field included_ranges fun(self: TSTree, include_bytes: true): Range6[]
 ---@field included_ranges fun(self: TSTree, include_bytes: false): Range4[]
 
+---@class TSQuery: userdata
+---@field inspect fun(self: TSQuery): TSQueryInfo
+
+---@class (exact) TSQueryInfo
```

Comment:
What does `(exact)` mean?

Reply:
https://luals.github.io/wiki/annotations/#class

This is a type owned by C, so no additional fields are not allowed to be written. It doesn't matter to have `(exact)` or not, I just wanted to make it more ensured.

Reply:
Oh nice! Didn't know about it :)